### DISCUSSION TITLE
Can't use `tokensOfOwner` function in ERC721AQueryable

### DISCUSSION THREAD

**Discussion by batmanbury** [hashlips-lab/nft-erc721-collection#459] [General]
I added a function to the contract file, which uses `tokensOfOwner` from ERC721AQueryable.sol. This is clearly imported at the top of the YourNftToken.sol file:

```
import "erc721a/contracts/extensions/ERC721AQueryable.sol";
```

And here is my new function, checking for ownership of a wallet:

```
  function checkNFTOwnership(address walletAddress) public view returns (uint256) {
    uint256[] memory tokenIds = tokensOfOwner(walletAddress);
    for (uint i = 0; i < tokenIds.length; i++) {
      uint256 tokenId = tokenIds[i];
      if (_exists(tokenId) && ownerOf(tokenId) == walletAddress) {
        // Check if the NFT belongs to this collection
        if (tokenId >= _startTokenId() && tokenId < _startTokenId() + maxSupply) {
          return tokenId;
        }
      }
    }
    return 0;
  }
```

But my VSCode editor is underlining this, and I am unable to deploy. All dependencies have been installed, and I have no errors otherwise. I clearly see the file containing `tokensOfOwners` inside node_modules after installation. I really have no idea why it would do this. Any ideas?
<img width="693" alt="image" src="https://user-images.githubusercontent.com/5461429/236100282-34b7af76-20f0-438b-921a-3bb55f4fcd36.png">
<img width="491" alt="Screen Shot 2023-05-03 at 10 47 35 PM" src="https://user-images.githubusercontent.com/5461429/236100509-72b09bf0-c950-475a-85f2-45624cf035d9.png">

**Comment by liarco** [comment]
Hi @batmanbury, I replied here: https://github.com/hashlips-lab/nft-erc721-collection/issues/460#issuecomment-1534256083
